### PR TITLE
Add plot method and data.py to NAIP

### PR DESCRIPTION
**Pull request by nilsleh**
Since RasterDatasets should have their own plot method per https://github.com/microsoft/torchgeo/issues/253, this PR adds a plot method as well as a data.py file to the NAIP dataset.

### CODE REVIEW

**Review by calebrob6 — DISMISSED** [review]

**Comment by calebrob6** [comment]
It looks like this is missing the data.py file, and the tests are not passing with the new test NAIP data

**Comment by calebrob6** [comment]
Maybe we can cut this down to 1PB with some optimizations?

image

File: tests/data/naip/data.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+import os
+import random
+
+import numpy as np
+import rasterio
+
+SIZE = 256
```
Comment (adamjstewart) [inline tests/data/naip/data.py:12]: 64 should work for all tests

File: torchgeo/datasets/naip.py [hunk 2]
```diff
@@ -42,3 +47,37 @@ class NAIP(RasterDataset):
     # Plotting
     all_bands = ["R", "G", "B", "NIR"]
     rgb_bands = ["R", "G", "B"]
+
+    def plot(  # type: ignore[override]
+        self,
+        sample: Dict[str, Tensor],
```
Comment (adamjstewart) [inline torchgeo/datasets/naip.py:53]:
```suggestion
        sample: Dict[str, Any],
```
See my comment on your other PR

**Review by adamjstewart — COMMENTED** [review]
Must be something from with the transform that's causing the file size to blow up to PBs?

File: tests/datasets/test_naip.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def test_or(self, dataset: NAIP) -> None:
     def test_plot(self, dataset: NAIP) -> None:
         query = dataset.bounds
         x = dataset[query]
-        dataset.plot(x["image"])
+        dataset.plot(x, suptitle="Test")
```
Comment (adamjstewart) [inline tests/datasets/test_naip.py:45]: Needs a `plt.close()`

**Review by adamjstewart — CHANGES_REQUESTED** [review]
We seem to still be allocating a ~1 PB numpy array somewhere, causing the tests to fail.

**Comment by adamjstewart** [comment]
Has anyone figured out what is going on with this giant numpy array?

**Comment by nilsleh** [comment]
> Has anyone figured out what is going on with this giant numpy array?

Could it have to do with the size of the bounds that is passed to index the dataset?

**Comment by adamjstewart** [comment]
That has to be it but I don't see anything fishy about the data. Maybe try printing the data length and dataset.bounds?

**Comment by adamjstewart** [comment]
P.S. To see the output of a print statement in a test file you'll need to use the `-s` flag:
```console
$ pytest -s tests/datasets/test_geo.py::TestRasterDataset::test_getitem_single_file
```

**Comment by nilsleh** [comment]
> That has to be it but I don't see anything fishy about the data. Maybe try printing the data length and dataset.bounds?

when running pytest on my ubuntu machine, I get:
`
BoundingBox(minx=0.0, maxx=64.0, miny=-63.0, maxy=1.0, mint=1541285999.999999, maxt=1559771999.999999)
torch.Size([4, 64, 64])`

**Comment by adamjstewart** [comment]
It's not the test_naip.py tests that are failing, it's the test_geo.py tests.

**Comment by nilsleh** [comment]
> it's the test_geo.py tests.

If I uncomment, the hardcorded crs of `crs = CRS.from_epsg(3005)` in the `test_geo.py` file, then the array is normal sized and the tests pass.

**Comment by adamjstewart** [comment]
The new test failure is because our NAIP data and Chesapeake data have no overlap.

**Comment by nilsleh** [comment]
> The new test failure is because our NAIP data and Chesapeake data have no overlap.

TJust for my understanding: The error I am seeing is that for the `IntersectionDataset` of `naip & chesapeake`, the `Bounding box is invalid: 'minx=1.7976931348623157e+308' > 'maxx=-1.7976931348623157e+308'`. Is that what happens when two datasets have no overlap?

I am not sure how to best proceed. If one now tunes the NAIP `data.py` to work with the Chesapeake bounds specifically, there might come a new test along that also needs to overlap with NAIP but then can't. What would you suggest to do?

**Comment by adamjstewart** [comment]
Correct, there's a bug in rtree where an empty index actually has invalid bounds. That's the error you're seeing. If you print the length of the dataset, you'll find that it's zero. 

I think it's fine to make sure that the bounding box for NAIP and Chesapeake match. The actual location of these datasets on earth doesn't matter, they just need to overlap. I don't envision a test being added that asserts that these particular datasets don't overlap.

File: tests/datasets/test_naip.py [hunk 4]
```diff
@@ -4,24 +4,27 @@
...
+    def dataset(self) -> NAIP:
         root = os.path.join("tests", "data", "naip")
         transforms = nn.Identity()  # type: ignore[no-untyped-call]
         return NAIP(root, transforms=transforms)
 
     def test_getitem(self, dataset: NAIP) -> None:
         x = dataset[dataset.bounds]
+        print(len(dataset))
+        print(dataset.bounds)
+        print(x["image"].shape)
```
Comment (adamjstewart) [inline tests/datasets/test_naip.py:27]: These print statements can be removed now

File: tests/datasets/test_geo.py [hunk 5]
```diff
@@ -149,10 +149,9 @@ class TestRasterDataset:
     @pytest.fixture(params=[True, False])
     def naip(self, request: SubRequest) -> NAIP:
         root = os.path.join("tests", "data", "naip")
-        crs = CRS.from_epsg(3005)
```
Comment (adamjstewart) [inline tests/datasets/test_geo.py:152]: It looks like removing this line is causing coverage to drop. We need at least one test where we sample from a CRS different than the file. Maybe we could change the CRS but also change the bounding box to be smaller than `dataset.bounds`? Or use a CRS with the same approximate CRS units as the files?

**Review by adamjstewart — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**